src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -197,14 +209,25 @@ const DateRangeInput: React.FC<
     [setCurrentRange, currentDateRange]
   );
 
-  const isDisabled = useCallback(
-    () =>
-      !currentDateRange[0] ||
-      !currentDateRange[1] ||
-      currentDateRange[0].isAfter(currentDateRange[1], withTime ? 'minute' : 'day'),
-    [currentDateRange]
-  );
-
+  const onClear = useCallback(() => setCurrentRange([undefined, undefined]), [setCurrentRange]);
+
+  const isDisabled = React.useMemo(() => {
+    if (areDatesUndefined(previousDateRange) && areDatesUndefined(currentDateRange)) {
+      return true;
+    }
+    if (currentDateRange[0] && !currentDateRange[1]) {
+      return true;
+    }
+
+    return currentDateRange.every((date, index) => {
+      return (
+        previousDateRange &&
+        previousDateRange[index] &&
+        // @ts-ignore
+        date?.isSame(previousDateRange[index], withTime ? 'minute' : 'day')
```

Comment:
optional: You can remove the `// @ts-ignore` with this:

```suggestion
        (date as Dayjs).isSame(previousDateRange[index], withTime ? 'minute' : 'day')
```

Reply:
Thanks for this one, fixed!